prusti-specs/src/specifications/preparser.rs

```diff
@@ -214,6 +235,44 @@ impl PrustiTokenStream {
         }
     }
 
+    fn parse_ghost_constraint(self) -> syn::Result<(TokenStream, Vec<NestedSpec<TokenStream>>)> {
+        let span = self.source_span;
+        let arguments = self.split(PrustiBinaryOp::Rust(RustOp::Comma), false);
+        let parsing_error = || {
+            error(span, "Invalid use of macro. Two arguments expected (a trait bound `T: A + B` and multiple specifications `[requires(...), ensures(...), ...]`)")
+        };
+
+        if arguments.len() < 2 {
+            return parsing_error();
+        }
+
+        // We interpret the last element of the arguments as the nested specs and everything before
+        // as the ghost constraints.
+        // This is due to the fact that the prusti token stream also splits on commas
+        // inside the ghost constraint.
+        let mut arg_iter = arguments.into_iter().peekable();
+        let mut trait_bounds_ts = PrustiTokenStream::empty(span);
+        let mut nested_specs: Option<Vec<NestedSpec<TokenStream>>> = None;
+        while let Some(mut arg) = arg_iter.next() {
+            if arg_iter.peek().is_some() {
+                // arg is part of ghost constraint
+                trait_bounds_ts.extend(arg);
+                trait_bounds_ts.tokens.push_back(PrustiToken::Token(TokenTree::Punct(Punct::new(',', Spacing::Alone))))
```

Comment:
Because the ghost constraints are `PrustiTokenStream`s and I first want to collect all ghost constraint into one single `PrustiTokenStream` before I can finally parse it with `parse_rust_only`.
When having something like `#[ghost_constraint(T: A<Foo, Baz = Bar>, ...)]`, there will be 3 PTS:
- (1): `T: A<Foo`
- (2): ` Baz = Bar>`
- (3): `...`
I think need to combine (1) and (2) into a new PTS first such that I do not get into unnecessary parsing errors.
Do you think I should immediately call `parse_rust_only` on (1) and (2) and combine the corresponding `TokenStream`s afterwards?

Reply:
Yes, I think parsing as "Rust only" and pushing into a `TokenStream` is nicer than extending the `PrustiTokenStream` API to support this.

Reply:
Done